### ISSUE TITLE
Batch Norm for 1D Convolutions

### ISSUE THREAD

**Issue by EdwardRaff** [fchollet/keras#4316]
I'm experience some odd behavior so I figured I would ask for clarification. 

For 2D data we need to give BatchNorm axis=1 to perform the correct normalization. For 1D data, we end up with (samples, timesteps, filters) for the data shape. Is the axis default of -1 still appropriate in this case?

**Comment by fchollet** [comment]
> For 2D data we need to give BatchNorm axis=1 to perform the correct normalization.

Only for the dimension ordering `(batch, filters, width, heigh)` (`dim_ordering='th'`). Otherwise -1 is the correct choice.

> Is the axis default of -1 still appropriate in this case?

Yes, because you want to normalize on the filters axis.

**Comment by EdwardRaff** [comment]
Thanks for the quick clarification. Wanted to make sure I wasn't pulling a stupid.
